src/main/java/com/onlinebookstore/onlinebookstore/mapper/ShoppingCartMapper.java

```diff
@@ -0,0 +1,15 @@
+package com.onlinebookstore.onlinebookstore.mapper;
+
+import com.onlinebookstore.onlinebookstore.config.MapperConfig;
+import com.onlinebookstore.onlinebookstore.dto.shoppingcart.ShoppingCartResponseDto;
+import com.onlinebookstore.onlinebookstore.model.ShoppingCart;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+
+@Mapper(config = MapperConfig.class, uses = CartItemMapper.class)
+public interface ShoppingCartMapper {
+    @Mapping(target = "userId", source = "user.id")
+    @Mapping(target = "cartItems", source = "cartItems", qualifiedByName = "fromItemsToDto")

```

Comment:
```suggestion
    @Mapping(target = "cartItems", source = "cartItems")
```
could you test it like that?